lib/compiler/passes/generate-bytecode.js

```diff
@@ -453,7 +549,7 @@ function generateBytecode(ast, options) {
 
       // istanbul ignore next Because we never generate invalid boundary type we cannot reach this branch
       default:
-        throw new Error(`Unknown boundary type "${boundary.type}" for the "repeated" node`);
+        throw unknownBoundary(boundary);
```

Comment:
Its there to let typescript verify that all the cases are covered. ie, if you add a new boundary type, but don't update the switch, ts will report an error on this line. 

So there's no change in the runtime behavior, but it will catch any errors at build time (and vscode will report the error as you type).

Reply:
Oh, I get it.  the `never` in the function is doing the work.  Nice.